components/patient-access-rules/ui/src/main/resources/PhenoTips/PatientAccessRightsManagement.xml

```diff
@@ -38,114 +38,32 @@
-      #set ($discard = $collaborator.put('type', $c.type))
-      #if ($c.type == 'user')
-        #set ($discard = $collaborator.put('name', $xwiki.getUserName("${c.user}", false)))
-      #else
-        #set ($discard = $collaborator.put('name', "$!{xwiki.getDocument($c.user).plainTitle}"))
-      #end
-      #set ($discard = $collaborator.put('accessLevel', $c.accessLevel.name))
-      #set ($discard = $collaboratorsData.add($collaborator))
-    #end
-    #set ($discard = $data.put('collaborators', $collaboratorsData))
-    $jsontool.serialize($data)
-  #end## Patient exists
-#elseif ("$!{request.xaction}" == 'update')
-  ## ===========================================================
-  ## Send update about access levels and collaborators
-  ## -----------------------------------------------------------
-  #set ($visibility = $_service.resolveVisibility("$!{request.visibility}"))
-  #set ($data = {})
-  #set ($result = true)
-  #set ($collaborators = $request.getParameterValues('collaborator'))
-  #set ($accessLevels = $request.getParameterValues('accessLevel'))
-  #set ($owner = "$!{request.get('owner')}")
-  #if ("$!{owner}" != '')
-    ## Check the validity of the new owner
-    #set ($invalidOwner = false)
-    #set ($validGroup = false)
-    #set ($validUser = false)
-    #set ($cannotTransfer = false)
-    #set ($ownerDocRef = $services.model.resolveDocument($owner))
-    #if ($xwiki.getDocument($ownerDocRef).getObject('PhenoTips.PhenoTipsGroupClass'))
-      #set ($validGroup = true)
-      ## Non-admin users can only transfer ownership to their own groups
-      ## Check to see if user has permission to transfer ownership
-      #set ($adminGroup = $owner + ' Administrators')
-      #if (!$xwiki.getUser().isUserInGroup($owner) &amp;&amp; !$xwiki.getUser().isUserInGroup($adminGroup) &amp;&amp; !$hasAdmin)
-        #set ($cannotTransfer = true)
+    #set ($canTransfer = $patientAccess.hasAccessLevel('owner'))
+    #set ($canManage = $patientAccess.hasAccessLevel('manage'))
+    #set ($owner = $patientAccess.getOwner())
+    #if ($owner.isGroup())
+      #set ($adminGroup = $xcontext.getDatabase() + ":Groups." + $owner.getUsername() + " Administrators")
+      #if (!$xwiki.getUser().isUserInGroup($adminGroup))
+        #set ($canTransfer = false)
       #end
-    #elseif ($services.query.xwql("from doc.object(XWiki.XWikiUsers) as user where user.name = :ownerName").bindValue("ownerName", "$owner").count() &gt; 0)
-      #set ($validUser = true)
     #end
-    #if (!($validUser || $validGroup))
-      #set ($invalidOwner = true)
+    #if ($hasAdmin)
+      #set ($canTransfer = true)
     #end
+    #set ($discard = $data.put('canTransfer', $canTransfer))
+    #set ($discard = $data.put('canManage', $canManage))
+    #set ($discard = $data.put('level', $patientAccess.getAccessLevel().getName()))
+    $jsontool.serialize($data)
   #end
-  #if (!$patient)
-    #set ($discard = $response.setStatus(400))## no patient : bad request
```

Comment:
In the new code i see an `#if ("$!patient.id" != '')` b ut see no `#else` which would set reply status to  `400 `? I presume  current code is always using the service correctly, but in general script service should be resilient and return error codes when needed